src/traces/scatter/index.js

```diff
@@ -855,3 +857,53 @@ scatter.hoverPoints = function(pointData, xval, yval, hovermode) {
 
     return [pointData];
 };
+
+var DESELECTDIM = 0.2;
+
+scatter.selectPoints = function(searchInfo, polygon) {
+    var cd = searchInfo.cd,
+        xa = searchInfo.xaxis,
+        ya = searchInfo.yaxis,
+        selection = [],
+        curveNumber = cd[0].trace.index,
+        marker = cd[0].trace.marker,
+        i,
+        di,
+        x,
+        y;
+
+    if(!marker) return; // TODO: include text and/or lines?
+
+    var opacity = Array.isArray(marker.opacity) ? 1 : marker.opacity;
+
+    if(polygon === false) { // clear selection
+        for(i = 0; i < cd.length; i++) cd[i].dim = 0;
+    }
+    else {
+        for(i = 0; i < cd.length; i++) {
+            di = cd[i];
+            x = xa.c2p(di.x);
+            y = ya.c2p(di.y);
+            if(polygon.contains([x, y])) {
+                selection.push({
+                    curveNumber: curveNumber,
+                    pointNumber: i,
+                    x: di.x,
+                    y: di.y
+                });
```

Comment:
We could certainly include those... and they can be useful at times. But yes, that was my thinking - when you want to serialize these objects, speed would suffer but also the extra pieces contain code references and DOM nodes that you can't serialize anyway.

But we should think carefully about what would be useful for people; for example, as it stands this requires people to keep track of what number each curve is, maybe we should include the trace name at least? And @cpsievert talked about some sort of a point key, which may be different from `pointNumber` I guess, but I don't know if there's a standard for this - perhaps we should define `trace.key` in our schema and use it for this (and then use this key to automatically select across subplots?)

And then after we think carefully about this, maybe we go back and clean up the hover event data...